apps/api/R/auth.R

```diff
@@ -41,6 +41,22 @@ validate_crypt_pass <- function(username, crypt_pass) {
   return(NA)
 }
 
+#* Check if the API key exists
+#* @param api_key API Key
+#* @return user details if apikey exists, else Invalid API Key
+#* @author Nihar Sanda
+validate_api_key <- function(api_key) {
+  res <- tbl(global_db_pool, "users") %>%
+    filter(apikey == api_key) %>%
+    collect()
+  
+  if (nrow(res) == 1) {
+    return(res)
+  }
+  
+  return(NA) 
+}
```

Comment:
Not sure how you're planning on having the users store the API key, but often packages have users put it in a .Renviron file.  Then, the `validate_api_key` can look for the key there by default.  An example: https://github.com/ropensci/webchem/blob/18e551609c8f1aa371f83aad1b281540e9df1066/R/chemspider.R#L22